### PR TITLE
api: avoid reinitializing exited containers on logs-only attach

### PR DESCRIPTION
**Pull request by Xiaowen-Yang**
#### Checklist

Ensure you have completed the following checklist for your pull request to be reviewed:

- [x] I have read and understood our [contributing guidelines](https://github.com/podman-container-tools/podman/blob/main/CONTRIBUTING.md) and will not have more than two open PRs as a new contributor.
- [x] PR description, commit message, and GitHub comments are human-written, per [LLM Policy](https://github.com/podman-container-tools/community/blob/main/LLM_POLICY.md)
- [x] Certify you wrote the patch or otherwise have the right to pass it on as an open-source patch by signing all
commits. (`git commit -s`). (If needed, use `git commit -s --amend`).  The author email must match
the sign-off email address. See https://github.com/podman-container-tools/podman/blob/main/CONTRIBUTING.md#sign-your-prs
for more information.
- [x] Referenced issues using `Fixes: #00000` in commit message (if applicable)
- [x] [Tests](https://github.com/podman-container-tools/podman/tree/main/test#readme) have been added/updated (or no tests are needed)
- [x] [Documentation](https://github.com/podman-container-tools/podman/blob/main/docs/README.md) has been updated (or no documentation changes are needed)
- [ ] All commits pass `make validatepr` (format/lint checks)
- [x] [Release note](https://github.com/kubernetes/community/blob/master/contributors/guide/release-notes.md) entered in the section below (or `None` if no user-facing changes) 
 
#### Does this PR introduce a user-facing change? 
Yes. 
 
 
```release-note 
This fixes externally observable container state and exit-code behavior for logs-only attach requests on exited containers. 
``` 
#### PR Description: 
- Problem: A logs-only attach on an already exited container triggered the reinitialization logic, changing the container state from exited to created and losing the original exit code. 
- Fix: Restrict container initialization to streamAttach requests. For logs-only requests, leave the container state unchanged. 
- Tests: Add a regression test that runs a container which exits with a non-zero exit code, performs a logs-only attach, verifies that the state remains exited and the exit code is preserved, and then confirms the exit code again through the wait API.

### CODE REVIEW

File: test/apiv2/26-containersWait.at [hunk 1]
```diff
@@ -39,6 +39,16 @@ t POST "containers/${CTR}/wait?condition=next-exit" 200 \
   .Error=null
 wait "${child_pid}"
 
+# Regression test for #29664: a logs-only attach to an exited container must
+# not reinitialize it or discard its exit code.
+t POST "containers/${CTR}/attach?stream=false&logs=true&stdout=true&stderr=true" 200
```
Comment (Honny1) [inline test/apiv2/26-containersWait.at:44]: Can you also add a test for: `stream=true&logs=true`.

**Review by Honny1 — CHANGES_REQUESTED** [review]
Can you please adjust the commit msg to this format:
```
<short subject line>

Fixes: #29664
Signed-off-by: Name Name <email@example.com>
```

File: test/apiv2/26-containersWait.at (hunk 1 above)
Comment (Xiaowen-Yang) [reply test/apiv2/26-containersWait.at:44]: > Can you also add a test for: `stream=true&logs=true`.

Thanks for the review. I’ve added a new test for stream=true&logs=true and updated the commit message according to the instructions. The API test also passes on my local machine. Could you please take another look when you have a chance? Thank you :)

**Review by Honny1 — APPROVED** [review]
LGTM

**Review by Luap99 — APPROVED** [review]
LGTM

I guess this is fine though I fail to understand why anyone would do a log only attach when the that is why the /containers/{name}/logs endpoint exists?
